monai/data/image_reader.py

```diff
@@ -1323,7 +1323,7 @@ def get_data(self, img: NrrdImage | list[NrrdImage]) -> tuple[np.ndarray, dict]:
             header = dict(i.header)
             if self.index_order == "C":
                 header = self._convert_f_to_c_order(header)
-            header[MetaKeys.ORIGINAL_AFFINE] = self._get_affine(i)
+            header[MetaKeys.ORIGINAL_AFFINE] = self._get_affine(header)
```

Comment:
I'm guessing the fix here is that the header being used has been passed through `self._convert_f_to_c_order` whereas this wasn't used in the original?

Reply:
Yes, and also affine is a copy of the original.
https://github.com/Project-MONAI/MONAI/pull/7415/files#diff-f9fa1fca729a2ee3040291d8ba9734458f2f8af4ef9b00fe7108ded2ec6006faL1333